src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1285,6 +1331,22 @@ public sealed override async Task<HttpResponseMessage> SendAsync(HttpRequestMess
                     // ISSUE 31315: Determine if/how to expose HTTP2 error codes
                     throw new HttpRequestException(SR.net_http_client_execution_error, e);
                 }
+                else if (e is OperationCanceledException)
+                {
+                    // If the operation has been canceled after the stream was allocated an ID, send a RST_STREAM.
+                    if (http2Stream != null && http2Stream.StreamId != 0)
+                    {
+                        http2Stream.Cancel();
+                    }
+
+                    OperationCanceledException oce = e as OperationCanceledException;
```

Comment:
You can delete this line and just change the above else if to:
```C#
else if (e is OperationCanceledException oce)
```

Reply:
I was't aware of that functionality. Thanks.